### ISSUE TITLE
[java] XPath "//MethodCall/AmbiguousName" not match

### ISSUE THREAD

**Issue by koalalam** [pmd/pmd#4946]
**Affects PMD Version:** 7.0.0

**Rule:** Custom XPath Rule to capture the use of old version of Apache Commons Collections

XPath 3.1
```
//ImportDeclaration[starts-with(@ImportedName,'org.apache.commons.collections.')]
| //MethodCall/AmbiguousName[starts-with(@Image,'org.apache.commons.collections.')]
```

**Description:**

The XPath expression works in Designer, but not in PMD.

**Code Sample demonstrating the issue:**

```java
import org.apache.commons.collections.CollectionUtils;

public class Test {
  public void Test() {
    Collection<String> emptyCollection = new ArrayList<>();
    boolean good4 = org.apache.commons.collections.CollectionUtils.isEmpty(emptyCollection);  // Line 6
  }
}
```

**Expected outcome:**

PMD should report a violation at line 6 but doesn't. This is a false-negative.

**Running PMD through:** *Gradle*

**Comment by koalalam** [comment]
> You should probably write
> 
> ```
> //MethodCall/TypeExpression[pmd-java:typeIs('org.apache.commons.collections.CollectionUtils')]
> ```

How about a prefix match against a package name? 

Below works in PMD v6.x:
```
//ImportDeclaration[starts-with(@ImportedName,'org.apache.commons.collections.')]
| //PrimaryPrefix/Name[starts-with(@Image, 'org.apache.commons.collections.')]
```

Is there a way to do it in PMD v7? Thank you very much!

**Comment by oowekyala** [comment]
Currently you can do that in XPath for the import but not for the method call. You can do it with a java rule though. `methodCall.getOverloadSelectionInfo().getMethodType().getSymbol().getPackageName()`

**Comment by koalalam** [comment]
Thanks @oowekyala. Any chance that we can have function e.g. `pmd-java:packageStartsWith` that we can use to build xpath rule?

**Comment by oowekyala** [comment]
I don't think a function would be useful here. We can easily add an attribute `@PackageQualifier` to `ASTClassType` though.

**Comment by koalalam** [comment]
@oowekyala Great! I assume we can then use `starts-with(@PackageQualifier, 'org.apache.commons.collections.')` like we did in PMD v6.

Please advise if raising a new issue for this feature request is needed. Thanks again!

**Comment by jsotuyod** [comment]
Yes, I'd recommend creating a new separate issue for that. Keeping track of all issues, when the details go into long threads is really difficult.
